pkg/chunk/schema.go

```diff
@@ -734,3 +774,47 @@ func (v10Entries) GetChunksForSeries(bucket Bucket, seriesID []byte) ([]IndexQue
 		},
 	}, nil
 }
+
+func (v10Entries) GetMetricsForSeries(_ Bucket, _ []byte) ([]IndexQuery, error) {
+	return nil, ErrNotSupported
+}
+
+// v11Entries builds on v10 but adds index entries for each series to store respective labels.
+type v11Entries struct {
+	v10Entries
+}
+
+func (s v11Entries) GetChunkWriteEntries(bucket Bucket, metricName string, labels labels.Labels, chunkID string) ([]IndexEntry, error) {
+	seriesID := labelsSeriesID(labels)
+	encodedThroughBytes := encodeTime(bucket.through)
+	data, err := jsoniter.ConfigFastest.Marshal(labels)
+	if err != nil {
+		return nil, err
+	}
+	entries := []IndexEntry{
+		// Entry for seriesID -> chunkID
+		{
+			TableName:  bucket.tableName,
+			HashValue:  bucket.hashKey + ":" + string(seriesID),
+			RangeValue: encodeRangeKey(encodedThroughBytes, nil, []byte(chunkID), chunkTimeRangeKeyV3),
+		},
+		// Entry for seriesID -> labels
+		{
```

Comment:
We should move this into `GetLabelWriteEntries` as those will be cached and deduped. I don't see a reason for this to be in `GetChunkWriteEntries`.

Reply:
done